### PR TITLE
[tool] Fix `filter-packages-to` when everything is changed

### PR DESCRIPTION
**Pull request by stuartmorgan**
`filter-packages-to` didn't correctly the handle the case where the set of target packages is empty, meaning that all packages should be tested. This broke it for cases such as changing a CI configuration file, making the filter not take effect.

### CODE REVIEW

File: script/tool/lib/src/common/package_command.dart [hunk 1]
```diff
@@ -436,16 +436,26 @@ abstract class PackageCommand extends Command<void> {
...
+    if (excludeAllButPackageNames != null &&
+        excludeAllButPackageNames.isNotEmpty) {
+      final List<String> sortedList = excludeAllButPackageNames.toList()
+        ..sort();
       print('--$_filterPackagesArg is excluding packages that are not '
           'included in: ${sortedList.join(',')}');
-      excludedPackageNames =
-          excludedPackageNames.union(packages.difference(filter));
+    }
+    // Utility to return true if a package that could be identified by any of
```
Comment (tarrinneal) [inline script/tool/lib/src/common/package_command.dart:451]: This is funky, but not a deal breaker imo

**Review by tarrinneal — APPROVED** [review]

File: script/tool/lib/src/common/package_command.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/common/package_command.dart:451]: I don't tend to use local functions very often, but they can be nice for small local code sharing.

Comment (tarrinneal) [reply script/tool/lib/src/common/package_command.dart:451]: I meant the comment itself, sorry. I thought my highlight would come through. 

> Utility to return true

**Review by ricardoamador — APPROVED** [review]

File: script/tool/lib/src/common/package_command.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/common/package_command.dart]: Oh, I see. Agreed; changed to just "Returns true if [...]".

**Pull request merged**